lm_eval/models/huggingface.py

```diff
@@ -337,6 +302,84 @@ def __init__(
                 f"Loglikelihood prefix token id used in evaluation: {self.prefix_token_id}"
             )
 
+    def _get_accelerate_args(
+        self,
+        parallelize: bool = None,
+        device_map: Optional[str] = "auto",
+        max_memory_per_gpu: Optional[Union[int, str]] = None,
+        max_cpu_memory: Optional[Union[int, str]] = None,
+        offload_folder: Optional[str] = "./offload",
+        gpus: Optional[int] = None,
+    ) -> dict:
+        """Returns the kwargs needed to apply `accelerate` in `AutoModel.from_pretrained`."""
+        num_local_processes = int(os.environ.get("LOCAL_WORLD_SIZE", 1))
+        num_machines = int(os.environ.get("WORLD_SIZE", 0)) // num_local_processes
+        if num_machines == 0:
+            eval_logger.info(
+                "We are not in a distributed setting. Setting model_parallel to False."
+            )
+            parallelize = False
```

Comment:
```suggestion
```

Only just caught this, but: I think this override is wrong?

Else, if someone tries to run pure `lm_eval --model_args parallelize=True` outside evaluate then it won't respect that `parallelize` value.

Reply:
I'm unsure I get what you mean in the "else" mention?

Reply:
If someone sets parallelize = True but does not have more than one GPU then it won't parallelize

Reply:
Or do you mean without accelerate?

Reply:
Fixed!